dhcpv6/ztp/parse_vendor_options.go

```diff
@@ -0,0 +1,60 @@
+package ztpv6
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/insomniacslk/dhcp/dhcpv6"
+)
+
+var (
+	errVendorOptionMalformed = errors.New("malformed vendor option")
+)
+
+// VendorData contains fields extracted from Option 17 data
+type VendorData struct {
+	VendorName, Model, Serial string
+}
+
+// ParseVendorData will try to parse dhcp6 Vendor Specific Information options data
+// looking for more specific vendor data (like model, serial number, etc).
+// If the options are missing we will just return an empty struct
+func ParseVendorData(packet dhcpv6.DHCPv6) (*VendorData, error) {
+	opt := packet.GetOneOption(dhcpv6.OptionVendorOpts)
+	if opt == nil {
+		return &VendorData{}, nil
```

Comment:
Should we return an error here and a nil `VendorData`? Returning an empty `VendorData` is probably not very useful.